### PR TITLE
feature: optional store capacity to prevent huge memory usage

### PR DESCRIPTION
**Pull request by dsalahutdinov**

### CODE REVIEW

**Comment by aderyabin** [comment]
Hi, @dsalahutdinov 
Thanks for PR.
My two remarks:
* Do not use `<<`. It uses more memory
* Add rotation option for capacity.

**Comment by dsalahutdinov** [comment]
hi, thx for review. what do you mean about rotation?

**Comment by aderyabin** [comment]
Well, it stops storing data on reaching a limit. I suggest to use like that:
```
Sniffer.config.store = {capacity: 1000, rotate: true}
```

It stores only last 1000 requests. On 1001 request it deletes first and add last to the end. It works like a queue

**Comment by dsalahutdinov** [comment]
Ok, i've got got it, nice feature. I'll implement it soon 😄

**Comment by dsalahutdinov** [comment]
I's done.

File: lib/sniffer.rb [hunk 1]
```diff
@@ -45,12 +45,33 @@ def reset!
...
+      else
+        push(data_item) unless overflow?
+      end
     end
 
     def logger
       config.logger
     end
+
+    private
```
Comment (aderyabin) [inline lib/sniffer.rb]: I think all these private methods should be moved from `Sniffer` because they are about `Sniffer.data`.

**Comment by dsalahutdinov** [comment]
Hi, i've extracted Data class. Please, take a look

**Comment by aderyabin** [comment]
Thanks

**Pull request merged**